### PR TITLE
Feature/loading bar

### PR DESCRIPTION
**Pull request by stevenbister**
### 🧹 Please check off the following before submitting:

- [ ] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [x] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [ ] Add milestone

---

## ☑ List all github issues that this PR will close:

N/a

## 📝 Description

Adds a loading bar to remix apps, using https://ricostacruz.com/nprogress/. I've added the stylesheet manually to each app so we can change colours and styles independently if we need to.

In ecommerce I've set it to appear at the top of the header rather than the top of the page as you can't see it if it matches the banner colour. I tried having it at the bottom of the header but it would inconsistently jump to the top of the body sometimes 🤷 Possible something to do with the header being position sticky 🤔 

## ⛳️ Current behaviour (updates)

N/a

## 🚀 New behaviour

Adds loading bar to remix apps

## 💣 Is this a breaking change (Yes/No):

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible cross-device impacts (iOS, Android)
- [ ] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

### Summary by CodeRabbit

**New Feature:**
- Added a global loading indicator to the `osc-academic-hub`, `osc-ecommerce`, and `osc-recruitment` apps. The loading indicator, powered by the NProgress library, provides visual feedback during page transitions and data fetching operations.
- Introduced new styles for the loading indicator in each app, enhancing the user interface.

**Refactor:**
- Optimized the codebase by removing unused imports and variables in the `osc-ecommerce` app.
- Improved code readability and maintainability by organizing CSS files into different layers using the `@layer` directive in Sass.

### CODE REVIEW

**Review by hi-im-si — CHANGES_REQUESTED** [review]
can we configure our own template so it keeps within our markup and naming conventions. the scss is hurting me 🗡️ 

https://github.com/rstacruz/nprogress#template

rough example:

```js
NProgress.configure({
    template: '<div class="c-page-loader" role="bar"><div class="c-page-loader__peg"></div></div><div class="c-page-loader__spinner" role="spinner"><div class="c-page-loader__spinner-icon"></div></div>'
});```

**Comment by stevenbister** [comment]
> can we configure our own template so it keeps within our markup and naming conventions. the scss is hurting me 🗡️
> 
> https://github.com/rstacruz/nprogress#template
> 
> rough example:
> 
> ```js
> NProgress.configure({
>     template: '<div class="c-page-loader" role="bar"><div class="c-page-loader__peg"></div></div><div class="c-page-loader__spinner" role="spinner"><div class="c-page-loader__spinner-icon"></div></div>'
> });```
> ```

Haha fair enough

File: packages/osc-academic-hub/app/styles/components/_n-progress.scss [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        box-sizing: border-box;
+        width: 18px;
+        height: 18px;
+        border: solid 2px transparent;
+        border-top-color: var(--color-primary);
+        border-left-color: var(--color-primary);
+        border-radius: 50%;
+        animation: nprogress-spinner 400ms linear infinite;
+    }
+}
```
Comment (hi-im-si) [inline packages/osc-academic-hub/app/styles/components/_n-progress.scss:53]: Dont think we are using the spinner so can probably remove this?

File: packages/osc-academic-hub/app/styles/components/_n-progress.scss [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+@keyframes nprogress-spinner {
+    0% {
+        transform: rotate(0deg);
+    }
+
+    100% {
+        transform: rotate(360deg);
+    }
+}
```
Comment (hi-im-si) [inline packages/osc-academic-hub/app/styles/components/_n-progress.scss:83]: If we are not using the spinner then probably wont need these styles either

**Review by hi-im-si — APPROVED** [review]

**Pull request merged**